### PR TITLE
Add a HeadingComponent to standardise headings

### PR DESCRIPTION
**Pull request by ethax-ross**
### Trello card

[Trello-3870](https://trello.com/c/EaKaoEaP/3870-address-heading-structure-on-promo-modules-including-find-module)

### Context

We have inconsistencies around how we display headings on the website. We also currently embed captions within the heading tag which is not accessible. In an effort to make headings more consistent going forward we're introducing a `HeadingComponent` and also moving the caption outside the heading as part of this.

### Changes proposed in this pull request

- Add heading component

Adds a `HeadingComponent` to help ensure consistent headings throughout the site, specifically when used with a caption.

- Remove subheadings/subtitles

These no longer appear to be referenced so we can remove the style and presentation code in the layout.

- Use HeadingComponent for captions

Anywhere we use a caption we want to render it via the `HeadingComponent` for consistency.

Remove all content from `left_section` of `PromoComponent` as we no longer want to display any here and it contained a caption.

- Update content guidance for headings

### Guidance to review

I started replacing all heading tags with the new component but the number of changes is enormous; I think it would be better to do this gradually as we go in and change content.

We can add more to the `HeadingComponent` to support box headings/highlighting/margins but I'll pick this up in a follow-up PR.

The headings updated so far are in the promo sections, the home page, the 'why teach' sections and the content page headings.

### CODE REVIEW

File: app/components/calls_to_action/promo/right_section.html.erb [hunk 1]
```diff
@@ -1,11 +1,10 @@
 <div class="promo__right">
-  <h2 class="heading-l">
-    <% if caption %>
-      <%= tag.div(caption, class: "caption-l caption--purple caption--bold") %>
-    <% end %>
-
-    <%= tag.div(heading) %>
-  </h2>
+  <%= render Content::HeadingComponent.new(
```
Comment (gemmadallmandfe) [inline app/components/calls_to_action/promo/right_section.html.erb:2]: @gazcarr These promo h2s and captions are now a bigger font size - just checking this is as expected from your perspective (fine with me either way)

eg h2 and caption: https://review-get-into-teaching-app-2975.london.cloudapps.digital/non-uk-teachers/international-qualified-teacher-status

eg longer h2: https://review-get-into-teaching-app-2975.london.cloudapps.digital/train-to-be-a-teacher/get-school-experience

File: app/views/content/non-uk-teachers/_why-teach.html.erb [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
   <div class="why-teach__container">
     <div class="why-teach__content">
-      <h2 class="heading-l">
-        <div class="caption-l caption--bold caption--purple">Why teach in England?</div>
-        Join a world class education system
-      </h2>
+      <%= render Content::HeadingComponent.new(
+        heading: "Join a world class education system",
+        tag: :h2,
+        caption: "Why teach in England?",
```
Comment (gemmadallmandfe) [inline app/views/content/non-uk-teachers/_why-teach.html.erb:7]: @gazcarr This caption (Why teach in England) is quite a bit smaller now - are you happy with this? I'm fine either way

https://review-get-into-teaching-app-2975.london.cloudapps.digital/non-uk-teachers

File: app/views/content/train-to-be-a-teacher/_why-teach.html.erb [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
   <div class="why-teach__container">
     <div class="why-teach__content">
-      <h2 class="heading-l">
-        <div class="caption-l caption--bold caption--purple">Why teach?</div>
-        Inspire the next generation
-      </h2>
+      <%= render Content::HeadingComponent.new(
+        heading: "Inspire the next generation",
+        tag: :h2,
+        caption: "Why teach?",
```
Comment (gemmadallmandfe) [inline app/views/content/train-to-be-a-teacher/_why-teach.html.erb:7]: @gazcarr This caption (Why teach?) is quite a bit smaller now - are you happy with this? I'm fine either way

https://review-get-into-teaching-app-2975.london.cloudapps.digital/train-to-be-a-teacher

File: app/views/content/funding-and-support/_why-teach.html.erb [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
   <div class="why-teach__container">
     <div class="why-teach__content">
-      <h2 class="heading-l">
-        <div class="caption-l caption--bold caption--purple">Why teach?</div>
-        Find out what teaching is really like
-      </h2>
+      <%= render Content::HeadingComponent.new(
+        heading: "Find out what teaching is really like",
+        tag: :h2,
+        caption: "Why teach?",
```
Comment (gemmadallmandfe) [inline app/views/content/funding-and-support/_why-teach.html.erb:7]: @gazcarr This caption (Why teach?) is quite a bit smaller now - are you happy with this? I'm fine either way

https://review-get-into-teaching-app-2975.london.cloudapps.digital/funding-and-support

**Review by gemmadallmandfe — COMMENTED** [review]
@ethax-ross I've had a look at the following:

* home page (difficult to tell with no captions currently!)
* promos (left a comment for @gazcarr)
* why teach (left some comments for @gazcarr)
* content page headings

All looks OK to me otherwise

File: app/views/content/funding-and-support/_why-teach.html.erb (hunk 4 above)
Comment (gazcarr) [reply app/views/content/funding-and-support/_why-teach.html.erb:7]: It should be bigger (roughly 28px) @ethax-ross

File: app/components/calls_to_action/promo/right_section.html.erb (hunk 1 above)
Comment (gazcarr) [reply app/components/calls_to_action/promo/right_section.html.erb:2]: the promo H2's (below the captions) should be roughly 36px @ethax-ross

**Comment by ethax-ross** [comment]
@gazcarr updated if you could recheck

**Review by gazcarr — COMMENTED** [review]
This is looking better

**Comment by gazcarr** [comment]
Screenshot 2022-11-28 at 16 17 02
is the gap still there @ethax-ross ?

**Comment by ethax-ross** [comment]
@gazcarr it looks like the review app didn't refresh after my last update, I've re-ran it and its updated now

**Comment by gemmadallmandfe** [comment]
Looks fine to me and I'm happy for this to be merged after @gazcarr has reviewed

**Comment by gazcarr** [comment]
@gemmadallmandfe @ethax-ross Looking good to me

**Review by gemmadallmandfe — APPROVED** [review]

**Pull request merged**